airgun/entities/acs.py

```diff
@@ -0,0 +1,823 @@
+            view.verify_ssl_toggle.click()
+
+        # Select CA certificate
+        if ca_cert is not None and view.verify_ssl_toggle.selected:
+            view.select_ca_cert.fill(ca_cert)
+
+        # Select manual authentication method
+        if manual_auth:
+            if (username is None) and (password is None):
+                raise ValueError(
+                    'At least one of username and password must be specified '
+                    'when using manual authentication!'
+                )
+
+            view.manual_auth_radio_btn.fill(True)
+            if username is not None:
+                view.username.fill(username)
+            if password is not None:
+                view.password.fill(password)
+
+        # Select content credentials authentication method
+        if content_credentials_auth:
+            if (ssl_client_cert is None) and (ssl_client_key is None):
+                raise ValueError(
+                    'At least one of ssl_client_cert and ssl_client_key '
+                    'must be specified when using content credentials authentication!'
+                )
+
+            view.content_credentials_radio_btn.fill(True)
+            if ssl_client_cert is not None:
+                view.ssl_client_cert.fill(ssl_client_cert)
+            if ssl_client_key is not None:
+                view.ssl_client_key.fill(ssl_client_key)
+
+        # Select no authentication method
+        if none_auth and not view.none_auth_toggle.selected:
+            view.none_auth_toggle.click()
+
+        # Save changes
+        view.edit_button.click()
+        self.close_details_side_panel()
+
+    def edit_products(
+        self,
+        acs_name_to_edit=None,
+        add_all=False,
+        remove_all=False,
+        products_to_add=None,
+        products_to_remove=None,
+    ):
+        """
+        Function that edits ACS products.
+
+        Args:
+            acs_name_to_edit (str): ACS name to be edited
+            add_all (bool): Whether to add all available options
+            remove_all (bool): Whether to remove all available options
+            products_to_add (str or list): List of products to add
+            products_to_remove (str or list): List of products to remove
+
+        Raises:
+            ValueError: If add_all and remove_all are True at the same time
+            ValueError: If add_all or remove_all are True at the same time
+                        with options_to_add or options_to_remove
+        """
+
+        if add_all and remove_all:
+            raise ValueError('add_all and remove_all cannot be True at the same time!')
+        if (add_all or remove_all) and (
+            products_to_add is not None or products_to_remove is not None
+        ):
+            raise ValueError(
+                'add_all or remove_all cannot be True at the same time with '
+                'options_to_add or options_to_remove!'
+            )
```

Comment:
I personally feel that this is an overhead for Airgun to handle these validations, Consider a case where someone wanted to test Airgun UI for invalid data, it won't happen because of validation.

Reply:
After a discussion, I will use a flag to enable/disable those checks.